src/ArmTemplates/Common/API/Clients/ApiSchemas/ApiSchemaClient.cs

```diff
@@ -28,7 +32,9 @@ public async Task<List<ApiSchemaTemplateResource>> GetApiSchemasAsync(string api
             var requestUrl = string.Format(GetAllApiSchemasRequest,
                this.BaseUrl, azSubId, extractorParameters.ResourceGroup, extractorParameters.SourceApimName, apiName, GlobalConstants.ApiVersion);
 
-            return await this.GetPagedResponseAsync<ApiSchemaTemplateResource>(azToken, requestUrl);
+            var apiSchemaTemplateResources = await this.GetPagedResponseAsync<ApiSchemaTemplateResource>(azToken, requestUrl);
+            this.commonTemplateResourceDataProcessor.ProcessData(apiSchemaTemplateResources);
```

Comment:
only 2 comments in the whole PR from my side:
1. maybe we can name it without `Common`? Like `ITemplateResourceDataProcessor`? What is the purpose of Common here?
2. since this line `this.commonTemplateResourceDataProcessor.ProcessData(<>TemplateResources);` exists in every single client, can we move it to the base class?

We can pass dataProcessors to base class and save them as protected fields. And expand `GetPagedResponseAsync()` with parameter `bool processResponse = true` and just do 
```
if (processResponse)
{
    this.templateResourceDataProcessor.ProcessData(response);
}
```

This will remove the same code from clients

Reply:
Thank you for the comments:
I do agree with renaming, it will be more readable, will update
Regarding the second one, it was not included into the base client because currently we have extractor/client which does not require the data processor (there is no name in retrieved resources) plus it will require to set suggested parameter to false for particular clients. Also keeping processing of the data separately allows customize the behaviour if needed without many changes, just via introducing new data processor. 
Thank you for suggestions and review! 
Open to discussion.

Reply:
Update: Renaming was done + removed and combined similar data processors

Reply:
looks great - thanks a ton!